aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -64,6 +64,33 @@ impl AggregatorID {
     }
 }
 
+/// Uniquely identifies each aggregator snapshot instance during the block execution.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct AggregatorSnapshotID {
+    pub id: u64,
+}
+
+/// Internal AggregatorSnapshot data structure.
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct AggregatorSnapshot {
+    // Describes a value of an aggregator.
+    value: u128,
+    // Describes a state of an aggregator.
+    state: AggregatorState,
+    // Describes an upper bound of an aggregator. If `value` exceeds it, the
+    // aggregator overflows.
+    // TODO: Currently this is a single u128 value since we use 0 as a trivial
+    // lower bound. If we want to support custom lower bounds, or have more
+    // complex postconditions, we should factor this out in its own struct.
+    limit: u128,
+    // Describes values seen by this aggregator. Note that if aggregator knows
+    // its value, then storing history doesn't make sense.
+    history: Option<History>,
+    // The AggregatorID of the aggregator from which the snapshot is taken.
+    base_aggregator: AggregatorID,
```

Comment:
originally you were thinking of adding base_aggregator to every Aggregator, and having AggregatorSnapshot only have base_aggregator. 
and then snapshot creates first a cloned aggregator, and then snapshot that points to it. 

just wondering what reasoning you had for adding all these in the AggregatorSnapshot, do you think it will be simpler?

Reply:
Linking with base aggregator is also easier for maintainability? Given that we plan to change quite a few things

Reply:
> Linking with base aggregator is also easier for maintainability? 

@georgemitenkov Do you mind expanding on this more? What do you mean by linking with base aggregator?

Reply:
I meant the same thing as Igor, having a `base_aggregator`

Reply:
> just wondering what reasoning you had for adding all these in the AggregatorSnapshot, do you think it will be simpler?

Earlier we had the notion of "derived aggregators". The goal at the time was to not introduce a new primitive and merge the notions of aggregator and derived aggregator. Over time, we separated out the notion of aggregators and aggregator snapshots quite a lot. Storing the aggregator snapshot information in rust code in an aggregator struct now seems more confusing to me.

Reply:
Sounds good, thanks for the explanation!